packages/connectivity/src/scp-cf/destination/destination-from-service.ts

```diff
@@ -471,11 +477,12 @@ Possible alternatives for such technical user authentication are BasicAuthentica
         )
       };
     }
-    // Case 2 Subscriber and provider account not the same OR custom jwt -> x-user-token  header passed to determine user and tenant in token service URL and service token to get the destination
+    // Case 2 Subscriber and provider account not the same OR custom JWT -> x-user-token header passed to determine user and tenant in token service URL and service token to get the destination
     const serviceJwt =
       origin === 'provider'
         ? this.providerServiceToken
-        : this.subscriberToken.serviceJwt!;
+        : // TODO: What is the meaning of this? Why do we assume this is defined. Technically, it might not be.
```

Comment:
@FrankEssenberger We are not ensuring that the service JWT is really not undefined. With `!` we are basically just assuming this, but it could still happen. Or not?

Reply:
If custom JWT, then origin has to be provider!